### DISCUSSION TITLE
[@types/ckeditor__ckeditor5-engine] Insufficient Callback Arguments for Observable

### DISCUSSION THREAD

**Discussion by mmichaelis** [DefinitelyTyped/DefinitelyTyped#56926] [Issues with a @types package] [answered]
The existing typing denotes two arguments (info and data) to be handed over to the callback:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/8980dbcb7d547ea04549bb8ef14369d40558d2db/types/ckeditor__ckeditor5-engine/src/view/document.d.ts#L32-L41

The official CKEditor documentation is more vague: It could be any function: [Observable.on](https://ckeditor.com/docs/ckeditor5/31.0.0/api/module_utils_observablemixin-Observable.html#function-on).

The given typing will fail when implementing focus tracking, as suggested in [Deep dive into focus tracking](https://ckeditor.com/docs/ckeditor5/31.0.0/framework/guides/deep-dive/ui/focus-tracking.html#checking-if-the-view-document-is-focused):

```javascript
editor.editing.view.document.on( 'change:isFocused', ( evt, data, isFocused ) => {
    console.log( `View document is focused: ${ isFocused }.` );
} );
```

I assume, a varargs argument of any type may be appropriate here.

[a comment by an automated account is not shown]

**Comment by fedemp** [comment, marked as answer]
@mmichaelis This has already been fixed on `master`.

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/905355a9745d99e7c2871bd6d3b4b7cbe2a78672/types/ckeditor__ckeditor5-utils/src/emittermixin.d.ts#L18-L24

Could you mark this as answered? :+1:
